Affiliated package blog (#238)
sunpy/sunpy.org: posts/2020/2020-08-17_affiliated_packages.rst

The What, Why, and How of SunPy Affiliated Packages
===================================================

.. post:: August 18 2020
   :author: Laura Hayes
   :tags: sunpy
   :category: Update

What are SunPy affiliated packages?
-----------------------------------
SunPy affiliated packages are well-maintained, open-source software packages that build upon or extend the functionality of the SunPy core library.
The main idea behind affiliated packages is to provide additional tools and functionality to the solar physics community that is considered outside the scope of the SunPy core library, but builds upon it.
Affiliated packages are aimed to integrate well with the SunPy ecosystem and together with the core library offer a diverse software ecosystem for which to perform solar data analysis.

> Affiliated packages are reviewed and registered to the SunPy project. 

Reviewer: ```suggestion
Affiliated packages are reviewed and registered with the SunPy project. 
```

There are two types of affiliated packages - *sponsored* and *non-sponsored*.
A sponsored affiliated package is an affiliated package that is maintained and overseen by the SunPy project, examples of these include `sunpy core <https://docs.sunpy.org/en/stable/>`_ and `ndcube <https://docs.sunpy.org/projects/ndcube/en/stable/>`_.

> Non-sponsored affiliated packages on the other hand remain under the control and maintenance of the original developers, but must meet a set of standards of the SunPy project.

Reviewer: does it read better like this?
```suggestion
Non-sponsored affiliated packages on the other hand remain under the control and maintenance of the original developers, but must meet a set of standards.
```

Package template
----------------
The SunPy project has developed a `package template <https://github.com/sunpy/package-template>`_ which is designed to help developers create new Python packages within the SunPy ecosystem.
The idea is that this template provides a simplified way to standardized packaging, testing, and documentation using the same framework as the SunPy core package.
The SunPy package template makes it easier for package developers to meet the standards required by SunPy to become an affiliated package!

How does a package become an affiliated package?
------------------------------------------------
To become a SunPy affiliated package, a set of criteria need to be met to ensure that affiliated packages provide useful functionality to the community at a standard of quality similar to the core SunPy package.
We now have a new review process for becoming an affiliated package, aimed at being open, approachable and streamlined!

The reviews are performed as an open process through GitHub issues on the `https://github.com/sunpy/sunpy.org` repository.
To submit a package for review to become an affiliated package, an issue should be opened on this repository.

> We now have an issue template for submitting a package for review to make it easier to provide all the required information and to start a dialogue about the process. 

Reviewer: You can link to the template using this link:

https://github.com/sunpy/sunpy.org/issues/new?assignees=Cadair&labels=Affiliated+Package+Review&template=new_affiliated_package.md

Once an issue is open, the Affiliate Package Liaison will check that the package meets the baseline requirements of an affiliated package.

> These include checking to see that it is compatible with the Sunpy `Code of Conduct <https://docs.sunpy.org/en/latest/code_of_conduct.html>`_,  has an appropriate licence, provides a Python interface, is on pypi, and is useful to the solar physics community.

Reviewer: ```suggestion
These include checking to see that it is compatible with the SunPy `Code of Conduct <https://docs.sunpy.org/en/latest/code_of_conduct.html>`_,  has an appropriate licence, provides a Python interface, is on PyPI, and is useful to the solar physics community.
```

For a submitted package to be accepted to affiliate package status, the package must be 'green' in **Functionality**, and one other category.
It must also not list any 'red' scores.

If the package in its current state does not pass the criteria, it can be listed as *Provisional* once it does not list 'red' in the **Functionality**, **Duplication** or **Community** criteria.
The idea is that the package can be listed as Provisional as it is working towards addressing the criteria for which it did not pass the review in. 

We would really like to encourage packages interested in becoming an affiliated SunPy package to please submit an issue and open up a review dialogue. 
We will have an Affiliated Package Liaison that will help you through each step of this process :).
Even if you are unsure about whether you want to submit a package, please feel free to open an issue and informally discuss your package. 

Reach out!
----------
If you are a developer of a package that you think fits nicely into the SunPy ecosystem and will of benefit to the solar physics community and want to chat to us about it please reach out!

> This can be of course regardless of how far along the package is - from concept to maturity! Join us our live chat `element channel <https://openastronomy.riot.im/#/room/#sunpy:openastronomy.org>`_ or join in on the SunPy weekly community meetings which occur on Wednesdays at 16:00 UTC and are hosted on `jitsi <https://sunpy.org/jitsi>`_.

Reviewer: ```suggestion
This can be of course regardless of how far along the package is - from concept to maturity! Join us our live chat `Matrix room <https://openastronomy.element.io/#/room/#sunpy:openastronomy.org>`_ or join in on the SunPy weekly community meetings which occur each Wednesday (see `our calendar <https://calendar.google.com/calendar?cid=ZzljOWVha2c5OGI1Y2JvZ2Q3bTV0YTZoOHNAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ>`_ for the time in your timezone) and are hosted on `jitsi <https://sunpy.org/jitsi>`_.
```

> 
> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```

Call to instrument/mission teams!
---------------------------------
We invite instrument and mission teams that are interested in developing a Python package to become a SunPy affiliated package. Integrating mission/instrument-specific software within the SunPy ecosystem will be of great benefit to the solar community and will allow for a simplified and unified process for users to perform solar data analysis. For example, a package may consist of instrument-specific calibration routines which would be outside the scope of SunPy core package, but could build upon the functionality provided by core such as the data containers and file readers. Developing a mission/instrument specific package as a sunpy affiliated package will allow the code to live and integrate within the SunPy ecosystem.

Furthermore, for mission/instrument teams that would like to begin developing specific python software, the SunPy project offers support through our package template and community resources for the package to be 'incubated' until it is at the level of affiliate status. We would really like to encourage instrument/mission teams to reach out!

> Package template

Author: yes good point - I've changed this now

How does a package become an affiliated package?
------------------------------------------------
To become a SunPy affiliated package, a set of criteria need to be met to ensure that affiliated packages provide useful functionality to the community at a standard of quality similar to the core SunPy package.
We now have a new review process for becoming an affiliated package, aimed at being open, approachable and streamlined!

The reviews are performed as an open process through GitHub issues on the `https://github.com/sunpy/sunpy.org` repository.
To submit a package for review to become an affiliated package, an issue should be opened on this repository.
We now have an `issue template <https://github.com/sunpy/sunpy.org/issues/new?assignees=Cadair&labels=Affiliated+Package+Review&template=new_affiliated_package.md>`_ for submitting a package for review to make it easier to provide all the required information and to start a dialogue about the process. 

Once an issue is open, the Affiliate Package Liaison will check that the package meets the baseline requirements of an affiliated package.

> These include checking to see that it is compatible with the SunPy `Code of Conduct <https://docs.sunpy.org/en/latest/code_of_conduct.html>`_,  has an appropriate licence, provides a Python interface, is on PyPi, and is useful to the solar physics community.

Reviewer: pedantry :stuck_out_tongue: 
```suggestion
These include checking to see that it is compatible with the SunPy `Code of Conduct <https://docs.sunpy.org/en/latest/code_of_conduct.html>`_,  has an appropriate licence, provides a Python interface, is on PyPI, and is useful to the solar physics community.
```
